backend/src/apiserver/server/job_server.go

```diff
@@ -65,6 +72,35 @@ func (s *JobServer) ListJobs(ctx context.Context, request *api.ListJobsRequest)
 	if err != nil {
 		return nil, util.Wrap(err, "Validating filter failed.")
 	}
+
+	if common.IsMultiUserMode() {
```

Comment:
Consider extract this section to a util function.

Reply:
This section is very similar to the code in ListRuns, but there is one difference for filter type "ExperimentUUID", only run has this, but job doesn't. 
So I'm thinking maybe we can delay the extraction until we added "ExperimentUUID" to job. By that time we would have a cleaner extracted util function reusable for both run and job, WDYT?

Reply:
what i meant is just extract this to a separate function, to improve the main function readability. it's ok we could do it later.

Reply:
Yes, I see your point. The run_server.go has similar problem. Once we have "ExperimentUUID" added in job, we can do a one-time extraction+merge change, instead of extract now and merge later.